### PR TITLE
Removes/Refactors /mob/unEquip

### LINKED ISSUE #15369: Duplication bugs
Sometimes when putting together a cyborg exoskeleton quickly (might happen with other items built by attacking with an item?), parts will not be used but will still be added to the frame. I was able to do this multiple times.

Embedded items will sometimes duplicate when they embed. The original item will be inside the person while the duplicate drops where it would of landed normally if it didn't embed. I somehow managed this with an energy sword, and was able to deconstruct both of them for research.

honkhonk

### LINKED ISSUE #20885: Taking something out of your backpack while on lava sets it on fire.
Because taking things out of your backpack apparently places said items on the floor for an instant before putting it in your hand if you're on lava when this happens (In a lavaboat/just immune to lava) the item will appear in your hand on fire and will eventually burn to ash.

### LINKED ISSUE #23314: Bunch of qdel loop item runtimes after explosions
Introduced by #22360 

Items are qdel'd, which causes them to drop on the exploding ground. Which causes them to qdel again.

### PR DESCRIPTION
**Pull request by Cyberboss**
The proc is still there, just renamed to `doUnEquip` and you should never call it. Overriding it is the same as before.

These 3 new mob procs replace it

- When you want to drop and item on the floor use `dropItemToGround(item, force)`
- When you want to change the loc of an item directly to another atom use `transferItemToLoc(item, newloc, force)`
- When you want to soon qdel an item or pass it as a param where it will be qdel'd or forceMove'd ise `temporarilyRemoveItemFromInventory(item, force)`

`temporarilyRemoveItemFromInventory` is a bit more dangerous because it deferences the item from the mob, but it still exists in the mob. Once it's called, it needs to have it's loc changed/be requipped or it will continue to just exist stuck inside the mob.

ALSO if you need to qdel an item YOU DON'T CALL ANY OF THESE BEFOREHAND. item/Destroy takes care of this itself.

Will update as I remember the shit I did

Fixes #20885 
Probably Fixes #15369
Fixes #23314

### CODE REVIEW

File: code/modules/paperwork/handlabeler.dm [hunk 1]
```diff
@@ -83,11 +83,9 @@
 /obj/item/weapon/hand_labeler/attackby(obj/item/I, mob/user, params)
 	..()
 	if(istype(I, /obj/item/hand_labeler_refill))
-		if(!user.unEquip(I))
-			return
 		user << "<span class='notice'>You insert [I] into [src].</span>"
 		qdel(I)
-		labels_left = initial(labels_left)
+		labels_left += initial(labels_left)
```
Comment (RemieRichards) [inline code/modules/paperwork/handlabeler.dm]: this lets you get more than ```initial(labels_left)``` labels, which Is not the intention.
for example the base has 30, so it's capped at a roll of 30.
yes it wastes the refill to use it at any less, but it's still a cap of 30, you've just broken the cap.

Comment (Cyberboss) [inline code/modules/paperwork/handlabeler.dm]: Ahh, thought it was an oversight

**Comment by Cyberboss** [comment]
@optimumtact DNM why?

**Comment by Cyberboss** [comment]
Is in testing now, remove DNM

**Comment by lzimann** [comment]
See #23310

**Comment by Cyberboss** [comment]
Still not fixed 1 moment

**Comment by Cyberboss** [comment]
Done, OP updated

**Comment by kevinz000** [comment]
YOU BROKE BACKPACKS

**Comment by Kromgar** [comment]
I DONT HAVE MY MINING VOUCHER WHAT HAVE YOU DONE or my internals box

**Comment by Cyberboss** [comment]
How....

**Comment by Cyberboss** [comment]
@kevinz000 be more specific dammit

**Comment by Kromgar** [comment]
This was testmerged. None of us had our starting internal boxes and none of our vouchers other jobs were missing gear too

**Comment by RemieRichards** [comment]
@Cyberboss what was I right about?

**Comment by Cyberboss** [comment]
Either in here or in IRC you said to make a function that will change the loc itself.

**Comment by RemieRichards** [comment]
oh yeah I did, I wasn't a fan of the allowing someone to forget to set the loc, I remember now.
Yeah 👍

**Comment by Cyberboss** [comment]
No complaints during the test merge, should be good

**Pull request merged**